### ISSUE TITLE
[Feature request] TalkBack support

### ISSUE THREAD

**Issue by dankeck** [yourealwaysbe/forkyz#66]
When using Forkyz with Android's built-in screen reader, TalkBack, it is possible to navigate the board using the arrow keys. However, the position and current value of the box is not announced,  so it's not very useful.

It would be great if users who are blind or have low vision could have the opportunity to use Forkyz and do crossword puzzles. One possible scheme could be:

- When the current clue changes at the top of the screen, this is an alert that is read to the user. Also the current filled in value is read ("blank" if no letters filled in, " C R O S S blank blank R D" if some are filled in.)
- When the user changes position within a clue, the value of the current box only is read.

The developers could play around with TalkBack to figure out what would be most intuitive.

Note: It is nice to see that the puzzle completion alert is already accessible with TalkBack.

**Comment by yourealwaysbe** [comment]
Thanks for getting in touch. Looks like adding the alerts should be straightforward. I'll do that today* and include it in Version 39 that i'm hoping to release soon (just giving it a few days to spot any bugs).

I'd made a small effort for accessibility a few years ago, but for some reason didn't manage to figure out how to use TalkBack. I obviously didn't try hard enough because it was easy enough to install it and set it up now.

If you have further suggestions to improve the workflow, don't hesitate to let me know.

(* since i'm spending today on the sofa watching the cycling :) )

**Comment by yourealwaysbe** [comment]
I realise there are two ways to implementing this. In short, is the current content description for the board view useful?

* I could set the clue display and the board as "live" so any changes to their description are announced. Then you'd get an announcement on a new clue. I could change the board description to contain the current word or the current box as you ask.

* I can use separate announcements out of the normal TalkBack flow. This works, but if you e.g. have the clue display focussed you get an announcement of the new clue display value plus the clue announced again via the separate announcement.

I'll go for dropping the current board view content description and replacing it with a description of the current box or word unless you tell me otherwise.

**Comment by yourealwaysbe** [comment]
I've set the clue view at the top to be "live" so that the current clue is announced.

Turns out using the content descriptions to carry information about the current selected box and so forth is bad practice. Making accessibility "announcements" is also discouraged, but i decided that was the most appropriate / least hacky solution.

I've also tried to follow the [guidance here](https://developer.android.com/guide/topics/ui/accessibility/custom-views#java). I've implemented what i think it says for the various board views, but as far as i can tell it makes no difference.

So, the main board view and the clue list announce clue changes and word/box contents as requested. The note page announces the board view boxes, but not the contents of the scratch and anagram boxes (the guidance i linked to should make that happen, but it isn't working for me).

I'll put it in Version 39 when i release it (soon). Let me know how it works out.

**Comment by yourealwaysbe** [comment]
A bit of a redesign: got rid of "live" views since it only really did half of what i wanted. Using announcements throughout now.

Also added "announce clue" features: optional ui button, accessibility action on board views, and when = pressed.

**Comment by yourealwaysbe** [comment]
I'll leave this issue open for feedback on the changes when they're out.
